src/qdrant_client/points.rs

```diff
@@ -24,6 +25,27 @@ impl Qdrant {
         &self,
         f: impl Fn(PointsClient<InterceptedService<Channel, TokenInterceptor>>) -> O,
     ) -> QdrantResult<T> {
+        if self.config.check_compatibility && self.is_compatible().is_none() {
+            let client_version = env!("CARGO_PKG_VERSION").to_string();
+            let server_version = match self.health_check().await {
+                Ok(info) => info.version,
+                Err(_) => "Unknown".to_string(),
+            };
```

Comment:
This only grabs the server version once.

We might have a cluster with multiple peers running different versions, or we might do a transparent version upgrade without re-instantiating the client.

It'd be probably better to include the version in server responses and parse that each time.

Wdyt?